calcom/cal.com
# Feature: Support redirecting to an external URL on successful booking

## What does this PR do?



Fixes #1457 #2088

## Type of change




## How should this be tested?



- Ensure only URL starting with http:// https:// are accepted even if a direct hit to mutation goes without browser validation.
- Upgrade Modal Box
    - Shown on Input click and Pro Badge Click. Pro Badge shown in Free account only. Trial and Pro doesn’t have it.
- Redirection Test
    - Automatic Redirect after 10 seconds
    - Cross Button to cancel automatic redirection. User stays on the success page and toast is dismissed
    - Continue button takes to external URL immediately.
    - Redirect URL can have query params or hash fragment and they are forwarded as is.
    - Mobile Device
        - Redirecting to {URL} is visible but timer isn’t there considering limited space.
- Team Event -  Redirect can be added and redirection works [Trial And PRO].
    - From **Trial** to **Free** downgrade Team Event keeps working(I would have expected it to stop working) and thus redirection keeps working with it.
    - From **Pro** to **Free** - Same thing
- User Event - Redirect can be added and redirection works.[Trial And PRO]
    - From **Trial** to **Free** downgrade - Redirect stops working but event booking is still working. User can’t edit the redirect URL but he would be able to see the already added URL which won’t work. Pro Upgrade Modal would be shown.
    - From **Pro** to **Free** downgrade - Same thing.

## Checklist



- I haven't checked if my changes generate no new warnings
- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes
